Added cli-app-with-nodejs (Workshop-Bounty)

aaryanporwal opened the pull request:
Hey 👋  
I made this workshop on CLI apps with node.js take a look!

Criteria:

- [x] Have instructive and compelling images, GIFs, and visuals
This workshop does contain instructive and compelling images, GIFs, and visuals.

- [x] Use repl.it or a similar service that allows members to begin coding with one click.
Repl.it is used for the entirety of the workshop.


- [x] Are very easy to hack on for long periods of time after completion
Many examples are listed on the end of the workshop as to how people can extend their project and learn more things.
There are appropriate demos, and linked an existing hack club's workshop to publish the app too.


- [x] Link unknown concepts to external sources.
Linked many complex concepts such as fetching API data and linked appropriate api documentation, and the documentation of the modules used
- [x]  End with at least 3 demos of how the project can be expanded, with live links and code links for each demo.
Have provided many demo examples to which people can refer and get an idea on how it can be expanded.
Including a dedicated "some other apps you can stem from here" section, but many ideas in between the workshop, so that one can start developing their own ideas while doing the workshop

 - [x] All workshops must be in English at native-level proficiency. Phrasing and grammar must be perfect.

- [x] All assets (images, PDFs, and any other binaries) must be hosted using the #cdn channel on the Hack Club Slack for posterity and to cut down on repository size.
All assets are hosted using the #cdn channel.

 - [x] Help club members produce a compelling and real project in under 20 minutes
The workshop is recorded and uploaded on the slack, here's the link: https://hackclub.slack.com/archives/C01A9V5LPD0/p1604053641015900

 You must be either a registered Hack Club leader or an active member of the Slack.
I'm an active community member for more than 6 months. 
My slack username is @ Aaryan Porwal

MatthewStanciu commented:
Woah!! This is so cool, and super informative. Also, that "Hacking" section is the most extensive hacking section I've ever seen—big props to you for that.

I honestly think this is almost ready to merge. A few things before I do, though:

- The image at the beginning of part 5 is broken
- I couldn't figure out how to use the weather app demo. When I ran the code as-is, I got an error (even though it should have defaulted to Portland?) and when I tried adding a city, it also didn't work. Take a look at the code and see if something's wrong there? And if the code is fine, maybe add instructions for running it in a comment

Once you've made these small changes, ping me on Slack and I'll re-review. 🚀

aaryanporwal commented:
Hey matthew thanks for reviewing, I've done the changes requested:

1. Replaced the old GIF

2. I fixed the weather-cli app (It requires API keys, and I just put mine, but suggested in comments to use your own by signing up)

MatthewStanciu commented:
Woohoooo! @aaryanporwal can you fill out the form at https://hack.af/bountypayment? I'll get your payment over ASAP
